### PR TITLE
Tech sanity tests

### PR DESCRIPTION
**Pull request by Areyana**
part of https://github.com/KasperskyLab/Kaspresso/issues/439

+ Old \Sgoogle.android\S can't cover all cases that we need, so i'm adding SystemSafetyPattern enum with multiple patterns, which anyone can extend.

+ Sanity tests that cover interceptors logic, except autoscroll compose that will be part of new compose features PR.

### CODE REVIEW

**Comment by Areyana** [comment]
related to #433 i'll remove system dialogs sanity tests for now

**Comment by adolfo706042525** [comment]
Do you need any help implementing specific features? I might be able to offer some advice.

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/systemsafety/SystemDialogSafetyPattern.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.kaspersky.kaspresso.systemsafety
+
+import java.util.regex.Pattern
+
+enum class SystemDialogSafetyPattern(val pattern: Pattern) {
```
Comment (Areyana) [inline kaspresso/src/main/kotlin/com/kaspersky/kaspresso/systemsafety/SystemDialogSafetyPattern.kt:5]: Old \\S*google.android\\S* can't cover all cases that we need, so i'm adding enum with multiple patterns, which anyone can extend.

File: samples/kaspresso-sample/src/main/kotlin/com/kaspersky/kaspressample/systemdialogs/SystemDialogsActivity.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    companion object {
+        private const val RESULT_CODE = 10
+    }
+
+    private lateinit var binding: ActivitySystemDialogsBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
```
Comment (Nikitae57) [inline samples/kaspresso-sample/src/main/kotlin/com/kaspersky/kaspressample/systemdialogs/SystemDialogsActivity.kt:20]: Same here

**Review by Nikitae57 — DISMISSED** [review]

File: samples/kaspresso-sample/src/main/kotlin/com/kaspersky/kaspressample/systemdialogs/SystemDialogsActivity.kt (hunk 2 above)
Comment (Areyana) [reply samples/kaspresso-sample/src/main/kotlin/com/kaspersky/kaspressample/systemdialogs/SystemDialogsActivity.kt]: deleted

File: samples/kautomator-sample/src/main/java/com/kaspersky/kaspresso/kautomatorsample/systemdialogs/SystemDialogsActivity.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    companion object {
+        private const val RESULT_CODE = 10
+    }
+
+    private lateinit var binding: ActivitySystemDialogsBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
```
[the remark this answers is not in the document]
Comment (Areyana) [reply samples/kautomator-sample/src/main/java/com/kaspersky/kaspresso/kautomatorsample/systemdialogs/SystemDialogsActivity.kt]: deleted

**Review by Nikitae57 — APPROVED** [review]

**Pull request merged**